client/src/graphs/svg/renderers/EpiRenderer.ts

```diff
@@ -182,4 +183,33 @@ export default class EpiRenderer extends SVGRenderer {
         .style('opacity', 0.2);
     });
   }
+
+  hideHighlight (): void {
+    const chart = (this as any).chart;
+
+    chart.selectAll('.node-ui').style('opacity', 1);
+    chart.selectAll('.edge').style('opacity', 1);
+  }
+
+  showHighlight (subgraph: SubgraphInterface): void {
+    const chart = (this as any).chart;
+    // FIXME: not very efficient
+    const nodes = subgraph.nodes;
+    const edges = subgraph.edges;
+    const nonNeighborNodes = chart.selectAll('.node-ui').filter(d => !nodes.map(node => node).includes(d.id));
+    nonNeighborNodes.style('opacity', 0.1);
+
+    const nonNeighborEdges = chart.selectAll('.edge').filter(d => !_.some(edges, edge => {
+      return edge.source === d.source && edge.target === d.target;
+    }));
+    nonNeighborEdges.style('opacity', 0.1);
+
+    const neighborNodes = chart.selectAll('.node-ui').filter(d => nodes.map(node => node).includes(d.id));
+    neighborNodes.style('opacity', 1);
+
+    const neighborEdges = chart.selectAll('.edge').filter(d => _.some(edges, edge => {
+      return edge.source === d.source && edge.target === d.target;
+    }));
+    neighborEdges.style('opacity', 1);
+  }
```

Comment:
In all those we can use `forEach` instead of `filter` to avoid going four times trought the same list and simplify the code.

i.e.
*From*
```JS
const nonNeighborNodes = chart.selectAll('.node-ui').filter(d => !nodes.map(node => node).includes(d.id));
nonNeighborNodes.style('opacity', 0.1);

const nonNeighborEdges = chart.selectAll('.edge').filter(d => !_.some(edges, edge => {
  return edge.source === d.source && edge.target === d.target;
}));
nonNeighborEdges.style('opacity', 0.1);

const neighborNodes = chart.selectAll('.node-ui').filter(d => nodes.map(node => node).includes(d.id));
neighborNodes.style('opacity', 1);

const neighborEdges = chart.selectAll('.edge').filter(d => _.some(edges, edge => {
  return edge.source === d.source && edge.target === d.target;
}));
neighborEdges.style('opacity', 1);
```
*To*
```JS
chart.selectAll('.edge').forEach(d => {
  const isNeighbour = edges.some(edge => {
    return edge.source === d.source && edge.target === d.target;
  });
  d.style('opacity', isNeighbour ? '1' : '0.1');
});

chart.selectAll('.node-ui').forEach(d => {
  const isNeighbour = nodes.map(node => node).includes(d.id);
  d.style('opacity', isNeighbour ? '1' : '0.1');
});
```